server/src/main/java/io/crate/execution/dml/ValueIndexer.java

```diff
@@ -66,8 +66,15 @@ default void collectSchemaUpdates(@Nullable T value,
      */
     default void updateTargets(Function<ColumnIdent, Reference> getRef) {}
 
+    /**
+     * @param storageIdentLeafName is a key in the source.
+     * If it's NULL, writing key must be skipped.
+     * For example, for array of primitives,
+     * we need to write key only once and inner primitive indexer should be writing only values.
+     */
     void indexValue(
         @Nullable T value,
+        @Nullable String storageIdentLeafName,
```

Comment:
Couldn't we add a default method impl here which handles the field writing and calls the existing `indexValue` without the field name?
This would avoid handling the same field write logic in every implementation...

Reply:
Thanks, looks simpler - reverted primitive type ValueIndexers. DynamicIndexer still has overwrites default as the whole point of this change was to change behavior of key writing.